src/pages/iou/request/step/IOURequestStepAmount.tsx

```diff
@@ -56,12 +56,12 @@ type IOURequestStepAmountProps = IOURequestStepAmountOnyxProps &
         transaction: OnyxEntry<OnyxTypes.Transaction>;
     };
 
-function getTaxAmount(transaction: OnyxEntry<OnyxTypes.Transaction>, policy: OnyxEntry<OnyxTypes.Policy>, newAmount: number) {
+function getTaxAmount(transaction: OnyxEntry<OnyxTypes.Transaction>, policy: OnyxEntry<OnyxTypes.Policy>, newAmount: number, currency: string) {
     if (!transaction?.amount) {
         return;
     }
     const transactionTaxCode = transaction?.taxCode ?? '';
-    const defaultTaxCode = TransactionUtils.getDefaultTaxCode(policy, transaction) ?? '';
+    const defaultTaxCode = TransactionUtils.getDefaultTaxCode(policy, transaction, currency) ?? '';
     const taxPercentage = TransactionUtils.getTaxValue(policy, transaction, transactionTaxCode ?? defaultTaxCode) ?? '';
```

Comment:
Just one issue I notice here. Otherwise this works fine.
Currently, the `tax code` and `tax amount` does not correlate correctly as displayed below. I think this is because we are giving preference to `transactionTaxCode` over `defaultTaxCode` here.
When a currency changes, is it intentional to reset the tax code to the default one (i.e. Foreign or workspace) even when the transaction tax code is set to a non-default one?

https://github.com/Expensify/App/assets/3004659/0e38e4bc-5f32-4972-b02f-696defbea914

Reply:
> When a currency changes, is it intentional to reset the tax code to the default one (i.e. Foreign or workspace) even when the transaction tax code is set to a non-default one?

Yes!

Reply:
Ok. Got it. Thanks. Now I know what is expected here.

Reply:
Thanks for clarifying!

> When a currency changes, is it intentional to reset the tax code to the default one (i.e. Foreign or workspace) even when the transaction tax code is set to a non-default one?

Also, this is behavior server side. When online if currency is changed, then server will update taxRate and hence taxAmount

Reply:
Just digged in a little deeper. Would the following solve this issue?

```
    const transactionCurrency = TransactionUtils.getCurrency(transaction);
    const taxCode = ((transactionCurrency !== currency) ?  defaultTaxCode : transactionTaxCode) ??  defaultTaxCode;
    const taxPercentage = TransactionUtils.getTaxValue(policy, transaction, taxCode) ?? '';
```

Reply:
It would, but we have this similar logic to get taxCode in `getDefaultTaxCode()` that is used by other functions too, so let's use it

Reply:
I think the logic within `getDefaultTaxCode()` and [here](https://github.com/Expensify/App/pull/42128#discussion_r1599736086) are not similar. The reasoning being that `getDefaultTaxCode()` returns one of the default tax codes whereas we want to choose to return either transaction’s tax code or default tax code [here](https://github.com/Expensify/App/pull/42128#discussion_r1599736086). But I may be missing your point. Once the code changes are done, I will take a look.

Reply:
Okay, i think I am getting the issue now